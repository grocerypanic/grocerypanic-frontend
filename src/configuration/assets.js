const Assets = {
  favicon: `${process.env.PUBLIC_URL}/favicon.ico`,
  nonBreakingSpace: "\u00A0",
  cart: `${process.env.PUBLIC_URL}/images/cart_desktop.jpg`,
  bill: `${process.env.PUBLIC_URL}/images/bill_desktop.jpg`,
  cart_mobile: `${process.env.PUBLIC_URL}/images/cart_mobile.jpg`,
  bill_mobile: `${process.env.PUBLIC_URL}/images/bill_mobile.jpg`,
};

export default Assets;
