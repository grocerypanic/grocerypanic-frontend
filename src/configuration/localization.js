import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { Strings } from "./strings";

const i18nConfig = {
  fallbackLng: {
    default: ["en"],
  },
  debug: true,
  ns: ["translation"],
  defaultNS: "translation",
  keySeparator: ".",
  interpolation: {
    escapeValue: false,
    formatSeparator: ",",
  },
  react: {
    wait: true,
  },
};

i18n.use(LanguageDetector).init({
  resources: {
    ...Strings,
  },
  ...i18nConfig,
});

export default i18n;
