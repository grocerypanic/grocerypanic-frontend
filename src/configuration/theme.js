export const primary = "#006600";
export const white = "#FFFFFF";
export const secondary = "#868e96";
export const tertiary = "#808080";
export const error = "#FF0000";
export const regular = "#000000";
export const success = "#5cb85c";
export const highlight = "blue";
export const selected = "red";
export const danger = "#d9534f";
export const dark = "#495057";
export const black = "#000000";
export const warning = "#ffc107";

export const ui = {
  alertTimeout: 2000,
  resizeTimeout: 1,
};

export const navbarSize = "50px";
export const tabSize = "26px";
export const itemAttributes = {
  height: 55,
  padding: 5,
  border: 10,
  spacing: 5,
  innerHeight: 40,
  placeHolderPadding: 12,
};

export const headerMobileThreshold = 480;
export const mobileThreshold = 480;

export const graph = {
  quantityLine: "rgb(2, 117, 216)",
  changeLine: "rgb(92, 184, 92)",
  topPadding: -12,
  holdingPatternHeight: "134px",
  chartContainerHeight: "150px",
  holdingPatternColor: "secondary",
  holdingPatternScale: 1,
  holdingPatternDivHeight: 20,
  holdingPatternAnimation: "grow",
};
