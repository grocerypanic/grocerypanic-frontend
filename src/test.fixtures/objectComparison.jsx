export const propCount = (hash, length) => {
  expect(Object.keys(hash).length).toBe(length);
};
