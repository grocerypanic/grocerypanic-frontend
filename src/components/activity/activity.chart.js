import Chart from "chart.js";

import { graph } from "../../configuration/theme";

export const renderChart = (translateFn, activity_last_two_weeks, item) => {
  const [quantities, changes] = generateChartData(
    activity_last_two_weeks,
    item
  );
  const labels = [translateFn("ItemActivity.GraphNow")];
  const dayMarker = translateFn("ItemActivity.GraphDayShortForm");

  activity_last_two_weeks.slice(1).forEach((_, index) => {
    const key = index * -1 - 1;
    labels.push(`${key}${dayMarker}`);
  });

  const ActivityChart = new Chart(generateCtx(), {
    type: "line",
    data: {
      labels,
      datasets: [
        {
          label: translateFn("ItemActivity.GraphChangeEvent"),
          borderColor: graph.changeLine,
          data: changes.map((o, index) => {
            return { y: o, x: index };
          }),
        },
        {
          label: translateFn("ItemActivity.GraphQuantity"),
          borderColor: graph.quantityLine,
          data: quantities.map((o, index) => {
            return { y: o, x: index };
          }),
        },
      ],
    },
    options: generateChartOptions(item, translateFn),
  });
  ActivityChart.render();
  return ActivityChart;
};

export const generateChartData = (activity_last_two_weeks, item) => {
  const changes = [];
  const quantities = [item.quantity];

  let quantity = item.quantity;

  activity_last_two_weeks.forEach((activity) => {
    changes.push(activity.change);
    quantity -= activity.change;
    quantities.push(quantity);
  });

  return [quantities, changes];
};

const generateChartOptions = (item, translateFn) => {
  return {
    scales: {
      yAxes: [
        {
          ticks: {
            maxTicksLimit: 9,
            precision: 0,
          },
        },
      ],
    },
    title: {
      text: item.name + ` (${translateFn("ItemActivity.GraphSubTitle")})`,
      display: true,
    },
    legend: {
      display: false,
      generateLabels: nullFunction,
    },
    layout: {
      padding: {
        left: 0,
        right: 5,
        top: graph.topPadding,
        bottom: 0,
      },
      margin: 0,
    },
    responsive: true,
    maintainAspectRatio: false,
  };
};

const generateCtx = () => {
  const ctx = document.getElementById("consumptionChart").getContext("2d");
  ctx.clearRect(0, 0, ctx.width, ctx.height);
  return ctx;
};

export const nullFunction = () => null;
