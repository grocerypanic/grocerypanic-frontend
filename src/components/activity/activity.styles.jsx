import styled from "styled-components";

import { graph, white, tertiary } from "../../configuration/theme";

export const Outline = styled.div`
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: ${tertiary};
  padding: 10px;
`;

export const ButtonBox = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const InnerBox = styled.div`
  display: flex;
  justify-content: center;
  align-items: left;
  flex-direction: row;

  background: ${white};
  margin: 5px;
  padding: 5px;

  .table {
    margin-bottom: 0px;
  }
  .chartBox {
    height: ${graph.chartContainerHeight};
    flex: 0 0 auto;
    width: 100%;
  }
`;
