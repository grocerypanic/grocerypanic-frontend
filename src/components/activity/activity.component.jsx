import React from "react";
import Table from "react-bootstrap/Table";
import { useTranslation } from "react-i18next";
import moment from "moment";
import PropTypes from "prop-types";

import HoldingPattern from "../holding-pattern/holding-pattern.component";

import { within7Days } from "../../util/datetime";

import { renderChart } from "./activity.chart";
import { graph } from "../../configuration/theme";

import { Paper } from "../../global-styles/containers";
import { Banner } from "../../global-styles/banner";
import { Outline, InnerBox } from "./activity.styles";

export const nullReport = {
  id: null,
  activity_first: null,
  usage_total: null,
  usage_avg_week: null,
  usage_avg_month: null,
  recent_activity: {
    user_timezone: null,
    usage_current_week: null,
    usage_current_month: null,
    activity_last_two_weeks: [],
  },
};

const ActivityReport = ({ item, activity_report }) => {
  const { t } = useTranslation();
  const ActivityChart = React.useRef();

  // state for activity

  React.useEffect(() => {
    if (!ready()) return;
    ActivityChart.current = renderChart(
      t,
      activity_report.recent_activity.activity_last_two_weeks,
      item
    ); // eslint-disable-line
    return () => {
      ActivityChart.current.destroy();
    };
  }, [activity_report]); // eslint-disable-line

  const calculateExpired = () => {
    if (item.expired > 0) return true;
    if (
      item.next_expiry_date.isBefore(moment()) &&
      item.next_expiry_quantity > 0
    )
      return true;
    return false;
  };

  const ready = () =>
    activity_report.recent_activity.activity_last_two_weeks.length > 0;

  return (
    <>
      <Paper>
        <Banner className="alert alert-success mb-2">
          {t("ItemActivity.Title")}
        </Banner>
        {calculateExpired() ? (
          <Banner className="alert alert-danger mb-2">
            {t("ItemActivity.RecommendExpiredItems")}
          </Banner>
        ) : null}
        {within7Days(item.next_expiry_date) && item.next_expiry_quantity > 0 ? (
          <Banner className="alert alert-warning mb-2">
            {`${item.next_expiry_quantity} ${t(
              "ItemActivity.RecommendExpiringSoon"
            )}`}
          </Banner>
        ) : null}
        <Outline>
          <InnerBox>
            <HoldingPattern
              condition={!ready()}
              color={graph.holdingPatternColor}
              animation={graph.holdingPatternAnimation}
              height={graph.holdingPatternHeight}
              scale={graph.holdingPatternScale}
              divHeight={graph.holdingPatternDivHeight}
            >
              <div className="chartBox">
                <canvas id="consumptionChart"></canvas>
              </div>
            </HoldingPattern>
          </InnerBox>
          <InnerBox>
            <Table striped bordered hover size="sm">
              <tbody>
                <tr>
                  <td>{t("ItemActivity.ConsumptionCurrentInventory")}</td>
                  <td>{item.quantity}</td>
                </tr>
                <tr>
                  <td>{t("ItemActivity.ConsumptionConsumedThisWeek")}</td>
                  <td data-testid="thisWeek">
                    {activity_report.recent_activity.usage_current_week}
                  </td>
                </tr>
                <tr>
                  <td>{t("ItemActivity.ConsumptionConsumedThisMonth")}</td>
                  <td data-testid="thisMonth">
                    {activity_report.recent_activity.usage_current_month}
                  </td>
                </tr>
                <tr>
                  <td>{t("ItemActivity.ConsumptionConsumedAvgWeek")}</td>
                  <td data-testid="avgWeek">
                    {activity_report.usage_avg_week}
                  </td>
                </tr>
                <tr>
                  <td>{t("ItemActivity.ConsumptionConsumedAvgMonth")}</td>
                  <td data-testid="avgMonth">
                    {activity_report.usage_avg_month}
                  </td>
                </tr>
              </tbody>
            </Table>
          </InnerBox>
        </Outline>
      </Paper>
    </>
  );
};

export default ActivityReport;

ActivityReport.propTypes = {
  item: PropTypes.object.isRequired,
  activity_report: PropTypes.object.isRequired,
};
