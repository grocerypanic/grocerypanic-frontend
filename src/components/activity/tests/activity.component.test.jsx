import "jest-canvas-mock";
import { render, cleanup, waitFor } from "@testing-library/react";
import moment from "moment";
import React from "react";
import Strings from "../../../configuration/strings";
import { graph } from "../../../configuration/theme";
import { AnalyticsContext } from "../../../providers/analytics/analytics.provider";
import { propCount } from "../../../test.fixtures/objectComparison";
import HoldingPattern from "../../holding-pattern/holding-pattern.component";
import { renderChart } from "../activity.chart";
import ActivityReport, { nullReport } from "../activity.component";

jest.mock("../activity.chart");
jest.mock("../../holding-pattern/holding-pattern.component");

const mockDestroy = jest.fn();
const mockEvent = jest.fn();
const mockAnalyticsSettings = { event: mockEvent, initialized: true };

HoldingPattern.mockImplementation(({ children }) => children);
renderChart.mockImplementation(() => jest.fn());

// Freeze Time
const OriginalDate = Date.now;
const mockDate = () =>
  (Date.now = jest.fn(() => new Date("2020-06-16T11:01:58.135Z")));

const mockItem = {
  expired: 0,
  id: 1,
  name: "Vegan Cheese",
  next_expiry_date: moment("2020-06-18").utc(),
  next_expiry_quantity: 1,
  preferred_stores: [1],
  price: "4.00",
  quantity: 1,
  shelf: 2,
  shelf_life: 5,
};

const mockActivityReport = {
  id: 1,
  activity_first: moment("2020-03-01").utc(),
  usage_total: 1,
  usage_avg_week: 0.17,
  usage_avg_month: 0.5,
  recent_activity: {
    user_timezone: "UTC",
    usage_current_week: 0,
    usage_current_month: 0,
    activity_last_two_weeks: [
      {
        date: "2020-06-18",
        change: 0,
      },
      {
        date: "2020-06-17",
        change: 0,
      },
      {
        date: "2020-06-16",
        change: 0,
      },
      {
        date: "2020-06-15",
        change: 0,
      },
      {
        date: "2020-06-14",
        change: 0,
      },
      {
        date: "2020-06-13",
        change: 0,
      },
      {
        date: "2020-06-12",
        change: 0,
      },
      {
        date: "2021-06-11",
        change: 0,
      },
      {
        date: "2021-06-10",
        change: 0,
      },
      {
        date: "2021-06-09",
        change: 0,
      },
      {
        date: "2020-06-08",
        change: 0,
      },
      {
        date: "2020-06-07",
        change: 0,
      },
      {
        date: "2020-06-06",
        change: 0,
      },
      {
        date: "2020-06-05",
        change: 0,
      },
    ],
  },
};

describe("Setup Environment", () => {
  let utils;
  let current;

  beforeEach(() => {
    mockDate();
    current = {
      item: { ...mockItem },
      activity_report: { ...mockActivityReport },
    };
  });

  afterEach(() => {
    Date.now = OriginalDate;
  });

  const renderHelper = (config) => {
    return render(
      <AnalyticsContext.Provider value={mockAnalyticsSettings}>
        <ActivityReport {...config} />
      </AnalyticsContext.Provider>
    );
  };

  describe("after report is fetched", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      renderChart.mockReturnValue({ destroy: mockDestroy });
    });

    afterEach(cleanup);

    describe("with a valid activity report", () => {
      beforeEach(() => {
        jest.clearAllMocks();
        utils = renderHelper(current);
      });

      it("should display average weekly consumption data", () => {
        expect(
          utils.queryByText(
            `${Strings.ItemActivity.ConsumptionConsumedAvgWeek}`
          )
        ).toBeTruthy();
        expect(utils.getByTestId("avgWeek").textContent).toBe(
          String(mockActivityReport.usage_avg_week)
        );
      });

      it("should display average monthly consumption data", () => {
        expect(
          utils.queryByText(
            `${Strings.ItemActivity.ConsumptionConsumedAvgMonth}`
          )
        ).toBeTruthy();
        expect(utils.getByTestId("avgMonth").textContent).toBe(
          String(mockActivityReport.usage_avg_month)
        );
      });

      it("should display current week's consumption data", () => {
        expect(
          utils.queryByText(
            `${Strings.ItemActivity.ConsumptionConsumedThisWeek}`
          )
        ).toBeTruthy();
        expect(utils.getByTestId("thisWeek").textContent).toBe(
          String(mockActivityReport.recent_activity.usage_current_week)
        );
      });

      it("should display current month's consumption data", () => {
        expect(
          utils.queryByText(
            `${Strings.ItemActivity.ConsumptionConsumedThisMonth}`
          )
        ).toBeTruthy();
        expect(utils.getByTestId("thisMonth").textContent).toBe(
          String(mockActivityReport.recent_activity.usage_current_month)
        );
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(renderChart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });

    describe("with expired items in inventory", () => {
      beforeEach(() => {
        current.item.expired = 9;
        jest.clearAllMocks();
        utils = renderHelper(current);
      });

      it("should render a warning about the expired items", () => {
        expect(utils.getByText(Strings.ItemActivity.RecommendExpiredItems));
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(renderChart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });

    describe("with items expiring soon in inventory", () => {
      beforeEach(() => {
        utils = renderHelper(current);
      });

      it("should render a warning items expiring soon", () => {
        expect(
          utils.getByText(
            `${current.item.next_expiry_quantity} ${Strings.ItemActivity.RecommendExpiringSoon}`
          )
        ).toBeTruthy();
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(renderChart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });

    describe("with no items expiring soon in inventory, but an old next_expiry_date set", () => {
      beforeEach(() => {
        jest.clearAllMocks();
        current.item.next_expiry_date = moment("2019-06-16");
        current.next_expiry_quantity = 0;
        utils = renderHelper(current);
      });

      it("should not render a warning items expiring soon", () => {
        expect(
          utils.queryByText(
            `${current.item.next_expiry_quantity} ${Strings.ItemActivity.RecommendExpiringSoon}`
          )
        ).toBeFalsy();
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(renderChart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });
  });

  describe("before transactions are fetched", () => {
    beforeEach(() => {
      current.activity_report = { ...nullReport };
      jest.clearAllMocks();
      renderChart.mockReturnValue({ destroy: mockDestroy });
    });

    describe("with items in the inventory", () => {
      beforeEach(() => {
        jest.clearAllMocks();
        utils = renderHelper(current);
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(true);
        expect(call.color).toBe(graph.holdingPatternColor);
        expect(call.animation).toBe(graph.holdingPatternAnimation);
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(graph.holdingPatternScale);
        expect(call.divHeight).toBe(graph.holdingPatternDivHeight);
        expect(call.children).toBeTruthy();
      });

      it("should not render the graph", async () => {
        expect(renderChart).toBeCalledTimes(0);
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });
  });
});
