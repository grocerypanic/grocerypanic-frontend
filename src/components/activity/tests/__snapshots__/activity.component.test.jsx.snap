// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment after report is fetched with a valid activity report should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Item Activity Report
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                used this week
              </td>
              <td
                data-testid="thisWeek"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                used this month
              </td>
              <td
                data-testid="thisMonth"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                avg used per week
              </td>
              <td
                data-testid="avgWeek"
              >
                0.17
              </td>
            </tr>
            <tr>
              <td>
                avg used per month
              </td>
              <td
                data-testid="avgMonth"
              >
                0.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment after report is fetched with expired items in inventory should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Item Activity Report
    </div>
    <div
      class="c1 alert alert-danger mb-2"
    >
      Expired Items Warning
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                used this week
              </td>
              <td
                data-testid="thisWeek"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                used this month
              </td>
              <td
                data-testid="thisMonth"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                avg used per week
              </td>
              <td
                data-testid="avgWeek"
              >
                0.17
              </td>
            </tr>
            <tr>
              <td>
                avg used per month
              </td>
              <td
                data-testid="avgMonth"
              >
                0.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment after report is fetched with items expirying soon in inventory should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Item Activity Report
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                used this week
              </td>
              <td
                data-testid="thisWeek"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                used this month
              </td>
              <td
                data-testid="thisMonth"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                avg used per week
              </td>
              <td
                data-testid="avgWeek"
              >
                0.17
              </td>
            </tr>
            <tr>
              <td>
                avg used per month
              </td>
              <td
                data-testid="avgMonth"
              >
                0.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment after report is fetched with no items expirying soon in inventory, but an old next_expiry_date set should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Item Activity Report
    </div>
    <div
      class="c1 alert alert-danger mb-2"
    >
      Expired Items Warning
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                used this week
              </td>
              <td
                data-testid="thisWeek"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                used this month
              </td>
              <td
                data-testid="thisMonth"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                avg used per week
              </td>
              <td
                data-testid="avgWeek"
              >
                0.17
              </td>
            </tr>
            <tr>
              <td>
                avg used per month
              </td>
              <td
                data-testid="avgMonth"
              >
                0.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment before transactions are fetched with items in the inventory should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Item Activity Report
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                used this week
              </td>
              <td
                data-testid="thisWeek"
              />
            </tr>
            <tr>
              <td>
                used this month
              </td>
              <td
                data-testid="thisMonth"
              />
            </tr>
            <tr>
              <td>
                avg used per week
              </td>
              <td
                data-testid="avgWeek"
              />
            </tr>
            <tr>
              <td>
                avg used per month
              </td>
              <td
                data-testid="avgMonth"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
